### ISSUE TITLE
gammy 0.9.63 does not compile on FreeBSD: undeclared identifier 'SHM_NORESERVE'

### ISSUE THREAD

**Issue by igalic** [Fushko/gammy#92]
```
meena@beastix /u/p/a/gammy (update/gammy)> make
===>  License GPLv3 accepted by the user
===>   gammy-0.9.63 depends on file: /usr/local/sbin/pkg - found
===> Fetching all distfiles required by gammy-0.9.63 for building
===>  Extracting for gammy-0.9.63
=> SHA256 Checksum OK for Fushko-gammy-v0.9.63_GH0.tar.gz.
===>  Patching for gammy-0.9.63
===>   gammy-0.9.63 depends on package: xorgproto>=0 - found
===>   gammy-0.9.63 depends on file: /usr/local/libdata/pkgconfig/x11.pc - found
===>   gammy-0.9.63 depends on file: /usr/local/libdata/pkgconfig/xext.pc - found
===>   gammy-0.9.63 depends on file: /usr/local/libdata/pkgconfig/xxf86vm.pc - found
===>   gammy-0.9.63 depends on file: /usr/local/lib/qt5/bin/moc - found
===>   gammy-0.9.63 depends on file: /usr/local/lib/qt5/bin/qmake - found
===>   gammy-0.9.63 depends on shared library: libGL.so - found (/usr/local/lib/libGL.so)
===>   gammy-0.9.63 depends on shared library: libQt5Core.so - found (/usr/local/lib/qt5/libQt5Core.so)
===>   gammy-0.9.63 depends on shared library: libQt5DBus.so - found (/usr/local/lib/qt5/libQt5DBus.so)
===>   gammy-0.9.63 depends on shared library: libQt5Gui.so - found (/usr/local/lib/qt5/libQt5Gui.so)
===>   gammy-0.9.63 depends on shared library: libQt5Widgets.so - found (/usr/local/lib/qt5/libQt5Widgets.so)
===>  Configuring for gammy-0.9.63
Info: creating stash file /usr/ports/accessibility/gammy/work/gammy-0.9.63/.qmake.stash
Project MESSAGE: Qt version: 5.15.2
Project MESSAGE: Release build
Project MESSAGE: Install path: /usr/local/bin
===>  Building for gammy-0.9.63
--- src/ui_mainwindow.h ---
--- src/ui_tempscheduler.h ---
--- src/ui_mainwindow.h ---
/usr/local/lib/qt5/bin/uic src/mainwindow.ui -o src/ui_mainwindow.h
--- src/ui_tempscheduler.h ---
/usr/local/lib/qt5/bin/uic src/tempscheduler.ui -o src/ui_tempscheduler.h
--- res/tmp/dspctl-xlib.o ---
c++ -c -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O3 -std=gnu++1z -Wall -Wextra -pthread -fPIC -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Iincludes -I/usr/local/include/qt5 -I/usr/local/include/qt5/QtWidgets -I/usr/local/include/qt5/QtGui -I/usr/local/include/qt5/QtDBus -I/usr/local/include/qt5/QtCore -Ires/tmp -I/usr/local/include/libdrm -Isrc -I/usr/local/include -I/usr/local/lib/qt5/mkspecs/freebsd-clang -o res/tmp/dspctl-xlib.o src/dspctl-xlib.cpp
--- res/tmp/main.o ---
c++ -c -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O3 -std=gnu++1z -Wall -Wextra -pthread -fPIC -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Iincludes -I/usr/local/include/qt5 -I/usr/local/include/qt5/QtWidgets -I/usr/local/include/qt5/QtGui -I/usr/local/include/qt5/QtDBus -I/usr/local/include/qt5/QtCore -Ires/tmp -I/usr/local/include/libdrm -Isrc -I/usr/local/include -I/usr/local/lib/qt5/mkspecs/freebsd-clang -o res/tmp/main.o src/main.cpp
--- res/tmp/dspctl-xlib.o ---
src/dspctl-xlib.cpp:170:92: error: use of undeclared identifier 'SHM_NORESERVE'
        shminfo.shmid = shmget(IPC_PRIVATE, img->bytes_per_line * img->height, IPC_CREAT | 0600 | SHM_NORESERVE);
                                                                                                  ^
1 error generated.
*** [res/tmp/dspctl-xlib.o] Error code 1

make[1]: stopped in /usr/ports/accessibility/gammy/work/gammy-0.9.63
1 error

make[1]: stopped in /usr/ports/accessibility/gammy/work/gammy-0.9.63
===> Compilation failed unexpectedly.
Try to set MAKE_JOBS_UNSAFE=yes and rebuild before reporting the failure to
the maintainer.
*** Error code 1

Stop.
make: stopped in /usr/ports/accessibility/gammy
meena@beastix /u/p/a/gammy (update/gammy) [1]>
```

**Comment by tcberner** [comment]
Moin moin 

FreeBSD does not support SHM_NORESERVE see shmget(2); so either there should be a build-system check for the support of it, or it could just be defined to 0 (as for example biology/star in the ports tree does already).


mfg Tobias

**Comment by igalic** [comment]
for reference:

- [`shmget(2)`](https://man.freebsd.org/shmget(2))
- [biology/star's patch](https://github.com/freebsd/freebsd-ports/blob/master/biology/star/files/patch-SharedMemory.cpp)

**Comment by igalic** [comment]
perhaps a better way to define that might be

```c
# if !defined(SHM_NORESERVE)
#  define SHM_NORESERVE 0
# endif

**Comment by Fushko** [comment]
I removed it altogether. Not very useful in my case and I'd rather not have linux-only extensions. I'll re-release 0.9.63 today with this change.

**Comment by igalic** [comment]
version numbers are cheap
just bump it to the next

**Comment by Fushko** [comment]
Fixed in 0.9.64
